lake/root.go

```diff
@@ -181,6 +184,25 @@ func (r *Root) lookupPoolByName(_ context.Context, name string) *PoolConfig {
 	return nil
 }
 
+func (r *Root) Lookup(ctx context.Context, nameOrID string) (ksuid.KSUID, error) {
```

Comment:
I might put a comment here that this is part of the DataAdapter interface. Also I might include a:
```
var _ proc.DataAdapter = (*Root)(nil)
```
somewhere at the top...

Reply:
I was a leading proponent of "Method implements implements package.Interface.Method" doc comments, but they tend to go stale as names change, so I'm now a leading opponent!

But I am a leading proponent of interface assertions, and I fully endorse adding this one.